File: StackUsingQueue.js
class StackUsingQueue{
    constructor(){
        this.q=[]
    }
    push(x){
        this.q.push(x)
        for(let i=0;i<this.q.length-1;i++){
            this.q.push(this.q.shift())
        }
    }
    pop(){
        return this.q.shift()
    }
    top(){
        return this.q[0]
    }
    empty(){
        return !this.q.length
    }  
}

obj=new StackUsingQueue()
obj.push(1)
obj.push(4)
obj.push(5)
obj.push(7)
console.log(obj.q)
obj.pop()
console.log(obj.q)
console.log(obj.top())

console.log(obj.empty())

File: StackUsingQueue.py
import collections
class StackUsingQueue:
    def __init__(self):
        self.queue =collections.deque()
    def push(self, x):
        self.queue.append(x)
        for i in range(len(self.queue)-1):
            self.queue.append(self.queue.popleft())
    def pop(self):
        return self.queue.popleft()
    def top(self):
        return self.queue[0]
    def empty(self):
        return not len(self.queue)
obj=StackUsingQueue()
obj.push(4)
obj.push(5)
obj.push(6)
print(obj.queue)
print(obj.pop())
print(obj.top())
obj.empty()
print(obj.queue)
obj.push(9)
print(obj.queue)